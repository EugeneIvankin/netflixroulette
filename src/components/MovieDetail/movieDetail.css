.movieDetail{
    background-color: #414141;
    height: auto;
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
}

h1, h2{
    font-family: Arial;
}

.headerMovieDetail{
    display: flex;
    height: 6vh;
    width: 85vw;
    margin-left: 7vw;
    justify-content: space-between;
    align-items: center;
}

.backwardBatton{
    height: 4vh;
    width: 10vw;
    display: inline;
    background-color: white;
    color: #f55263;
    font-size: 2vw;
    cursor: pointer;
}

.movieFigure{
    float: left;
    width: 30vw;
    height: 46vw;
    margin-top: 2vh;
    margin-left: 7vw;
    margin-bottom: 7vh;
}

.movieInfo{
    width: 50vw;
    margin: 3vh 7vw 0 3vw;
    float: right;
    cursor: default;
}

.movieName{
    width: 70%;
    font-size: 3rem;
    color: #f55263;
    margin-bottom: 0;
    margin-top: 0;
}

.movieMark{
    border-style: solid;
    border-radius: 50%;
    width: auto;
    height: auto;
    text-align: center;
    color: #b5e584;
    border-width: 1px;
    font-weight: 100;
    position: absolute;
    top: 8vh;
    font-size: 2vw;
    padding: 1%;
    right: 10vw;
}

.movieYear{
    display: block;
    margin-top: 6vh;
    color: white;

}

.movieDurability{
    color: white;
}

.movieStory, .movieDirector, .movieCast{
    font-weight: 100;
    color: #c8c8c8;
}

.movieStory{
    color: #d7d7d7;
    margin-bottom: 7vh;
}

