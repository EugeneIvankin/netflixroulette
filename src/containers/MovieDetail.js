import { connect } from 'react-redux';
import * as searchActions from '../actions/SearchActions';
import { bindActionCreators } from 'redux';
import * as movieDetailAction from '../actions/MovieDetailActions';
import MovieDetailPage from '../components/MoviePages/MovieDetailPage';


function mapStateToProps(state) {
  return {
    movies: state.searchMovies,
    movie: state.movieDetail
  };
}

function mapDispatchToProps(dispatch) {
  return {
    sortMovies: bindActionCreators(searchActions.sortMovies, dispatch),
    getMovieInfo: bindActionCreators(movieDetailAction.getMovieInfo, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieDetailPage);