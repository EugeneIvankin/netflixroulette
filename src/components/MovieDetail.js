import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../styles/movieDetail.css';

export default class MovieDetail extends Component {

    constructor(props){
        super(props);
        this.backToSearch = this.backToSearch.bind(this);
    }

    backToSearch() {
        this.props.history.push(`/`);
    }

    render() {
        const {movie} = this.props;

        if(!movie){
            return null;
        }

        return <main className="movieDetail">

            <header className="headerMovieDetail">
                <p className="logo">netflixroulette</p>
                <button className="backwardBatton" onClick={() => {this.backToSearch(movie.movieName)}}><h2 className="backwardText">SEARCH</h2></button>
            </header>

            <div className="foundMovie">
                <figure className="movieFigure">
                    <img src={`https://image.tmdb.org/t/p/w640${movie.moviePoster}`} className="movieImage"/>
                </figure>

                <div className="movieInfo">
                    <h1 className="movieName">{movie.movieName}</h1>
                    <h2 className="movieMark">{movie.movieLevel}</h2>
                    <h2 className="movieYear">{movie.movieYear}</h2>
                    <h2 className="movieDurability">{movie.movieDurability} min.</h2>
                    <h2 className="movieStory">{movie.movieStory}</h2>
                </div>
            </div>
        </main>
    }
}

MovieDetail.propTypes = {
    movie: PropTypes.object
};