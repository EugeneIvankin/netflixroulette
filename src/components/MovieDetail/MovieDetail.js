import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './movieDetail.css';

export default class MovieDetail extends Component {

    backToSearch = () => {
        this.props.history.push(`/`);
    }

    render() {
        const {movie} = this.props;

        if(!movie){
            return null;
        }

        return <main className="movie-detail">
            <header className="header-movie-detail">
                <p className="logo">netflixroulette</p>
                <button className="backward-batton" onClick={this.backToSearch}>SEARCH</button>
            </header>
            <div className="foundMovie">
                <figure className="movie-figure">
                    <img src={`https://image.tmdb.org/t/p/w640${movie.moviePoster}`}/>
                </figure>

                <div className="movie-info">
                    <h1 className="movie-name">{movie.movieName}</h1>
                    <h2 className="movie-mark">{movie.movieLevel}</h2>
                    <h2 className="movie-year">{movie.movieYear}</h2>
                    <h2 className="movie-durability">{movie.movieDurability} min.</h2>
                    <h2 className="movie-story">{movie.movieStory}</h2>
                </div>
            </div>
        </main>
    }
}

MovieDetail.propTypes = {
    movie: PropTypes.object
};