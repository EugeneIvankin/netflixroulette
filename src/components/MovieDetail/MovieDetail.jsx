import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './movieDetail.css';

export default class MovieDetail extends Component {
  backToSearch = () => {
    this.props.history.push('/');
  };

  render() {
    const { movie } = this.props;

    if (!movie) {
      return null;
    }

    return (
      <main className="movie-detail">
        <header className="header-movie-detail">
          <p className="logo">netflixroulette</p>
          <button className="backward-batton" onClick={this.backToSearch}>SEARCH</button>
        </header>
        <div className="foundMovie">
          <figure className="movie-figure">
            <img src={`https://image.tmdb.org/t/p/w640${movie.moviePoster}`} alt="Poster" />
          </figure>
          <div className="movie-info">
            <h1 className="movie-name">{movie.movieName}</h1>
            <h2 className="movie-mark">{movie.movieLevel}</h2>
            <h2 className="movie-year">{movie.movieYear}</h2>
            <h2 className="movie-durability">{movie.movieDurability} min.</h2>
            <h2 className="movie-story">{movie.movieStory}</h2>
          </div>
        </div>
      </main>
    );
  }
}

MovieDetail.propTypes = {
  movie: PropTypes.shape({
    moviePoster: PropTypes.string,
    movieName: PropTypes.string,
    movieLevel: PropTypes.number,
    movieYear: PropTypes.string,
    movieDurability: PropTypes.number,
    movieStory: PropTypes.string,
  }),
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

MovieDetail.defaultProps = {
  movie: {
    moviePoster: '',
    movieName: '',
    movieLevel: 0,
    movieYear: '',
    movieDurability: 0,
    movieStory: '',
  },
};
