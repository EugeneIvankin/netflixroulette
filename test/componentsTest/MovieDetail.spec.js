import React from 'react';
import { shallow } from 'enzyme';
import MovieDetail from '../../src/components/MovieDetail/MovieDetail'

describe('<MovieDetail/>', () => {

    const foundedMovie = {
        movieId: '1',
        movieName: 'Taxi',
        moviePoster: 'url',
        movieYear: '2017',
        movieLevel: '5.5'
    };

    it ('should render MovieDetail', () => {
        const renderedComponent = shallow(
          <MovieDetail movie={foundedMovie}/>
        );
        expect(renderedComponent).toMatchSnapshot();
    });

    it ('should call backToSearch', () => {
        const history = {
            push: jest.fn()
        };

        const renderedComponent = shallow(
          <MovieDetail history={history}/>
        );
        renderedComponent.instance().backToSearch();
        expect(history.push).toBeCalledWith('/');
    });

});