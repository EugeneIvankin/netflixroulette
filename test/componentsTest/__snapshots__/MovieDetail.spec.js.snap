// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetail/> should render MovieDetail 1`] = `
<main
  className="movie-detail"
>
  <header
    className="header-movie-detail"
  >
    <p
      className="logo"
    >
      netflixroulette
    </p>
    <button
      className="backward-batton"
      onClick={[Function]}
    >
      SEARCH
    </button>
  </header>
  <div
    className="foundMovie"
  >
    <figure
      className="movie-figure"
    >
      <img
        alt="Poster"
        src="https://image.tmdb.org/t/p/w640url"
      />
    </figure>
    <div
      className="movie-info"
    >
      <h1
        className="movie-name"
      >
        Taxi
      </h1>
      <h2
        className="movie-mark"
      >
        5.5
      </h2>
      <h2
        className="movie-year"
      >
        2017
      </h2>
      <h2
        className="movie-durability"
      >
         min.
      </h2>
      <h2
        className="movie-story"
      />
    </div>
  </div>
</main>
`;
